### PR TITLE
Add support for POST application/x-www-form-urlencoded

### PR DESCRIPTION
**Pull request by alamcordeiro**
continue "simplest and < 100 lines"

use passing form object in opts.
```
get({url: 'http://...', form: {foo: 'bar'}}, () => {})
```

### CODE REVIEW

**Comment by yciabaud** [comment]
Thanks for this contribution @alamcordeiro! 
It looks good and I believe this can be useful but can you consider adding a test case for this?

**Comment by alamcordeiro** [comment]
Thanks, test added now

File: index.js [hunk 1]
```diff
@@ -63,14 +65,12 @@ module.exports.concat = function (opts, cb) {
...
       var data = Buffer.concat(chunks)
-      if (opts.json) {
-        try {
-          data = JSON.parse(data.toString())
-        } catch (err) {
-          return cb(err, res, data)
-        }
+      if (!opts.json) return cb(null, res, data)
+      try {
+        cb(null, res, JSON.parse(data.toString()))
```
Comment (feross) [inline index.js]: It's a very bad idea to call a callback function from within a try block. This means that if the user's function throws an exception than `simple-get` will catch and handle it.

Can you revert this part of this PR?

File: test/basic.js [hunk 2]
```diff
@@ -508,6 +509,36 @@ test('post (json body)', function (t) {
...
+    var port = server.address().port
+    var opts = {
+      method: 'POST',
+      url: 'http://localhost:' + port,
+      form: formData
+    }
+    get.concat(opts, function (err, res, data) {
+      t.error(err)
+      t.equal(res.statusCode, 200)
+      t.equal(JSON.stringify(qs.parse(data.toString())), JSON.stringify(formData))
```
Comment (feross) [inline test/basic.js]: Could have just used `t.deepEqual` and skipped the `JSON.stringify` calls.

File: index.js [hunk 3]
```diff
@@ -15,13 +16,14 @@ function simpleGet (opts, cb) {
   if (opts.headers == null) opts.headers = {}
   if (opts.maxRedirects == null) opts.maxRedirects = 10
 
-  var body = opts.json ? JSON.stringify(opts.body) : opts.body
+  var body = opts.json ? JSON.stringify(opts.body) : opts.form ? qs.stringify(opts.form) : opts.body
```
Comment (feross) [inline index.js]: I assume this option is modeled after the same option in `request`. If `opts.form` is already a query string then we should not call `qs.stringify`.

File: index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ var https = require('https')
 var once = require('once')
 var unzipResponse = require('unzip-response') // excluded from browser build
 var url = require('url')
+var qs = require('querystring')
```
Comment (feross) [inline index.js]: Let's keep the name `querystring`

**Comment by alamcordeiro** [comment]
Thanks for tips.
This is the goal? https://github.com/alamcordeiro/simple-get/commit/21aba210535032d8f23cfd7815defdec0d9fd6c2

**Comment by alamcordeiro** [comment]
@feross

**Pull request merged**